src/KafkaW/Consumer.h

```diff
@@ -1,77 +1,38 @@
 #pragma once
 
 #include "BrokerSettings.h"
-#include "Message.h"
-#include <functional>
-#include <librdkafka/rdkafka.h>
-#include <memory>
+#include "ConsumerMessage.h"
+#include "KafkaEventCb.h"
+#include "helper.h"
+#include <vector>
 
 namespace KafkaW {
 
 class ConsumerInterface {
 public:
   ConsumerInterface() = default;
   virtual ~ConsumerInterface() = default;
-  virtual void addTopic(std::string Topic) = 0;
-  virtual std::unique_ptr<Message> poll() = 0;
+  virtual void addTopic(const std::string &Topic) = 0;
+  virtual std::unique_ptr<ConsumerMessage> poll() = 0;
 };
 
 class Consumer : public ConsumerInterface {
 public:
-  explicit Consumer(BrokerSettings Settings);
+  explicit Consumer(BrokerSettings &opt);
   Consumer(Consumer &&) = delete;
   Consumer(Consumer const &) = delete;
   ~Consumer() override;
-  void init();
-  void addTopic(std::string Topic) override;
-  std::unique_ptr<Message> poll() override;
-  std::function<void(rd_kafka_topic_partition_list_t *plist)>
-      on_rebalance_assign;
-  std::function<void(rd_kafka_topic_partition_list_t *plist)>
-      on_rebalance_start;
-  rd_kafka_t *RdKafka = nullptr;
+  void addTopic(const std::string &Topic) override;
+  std::unique_ptr<ConsumerMessage> poll() override;
 
+protected:
+    std::shared_ptr<RdKafka::KafkaConsumer> KafkaConsumer;
+    virtual std::vector<int32_t> getTopicPartitionNumbers(const std::string &Topic);
 private:
+    RdKafka::Conf* Conf;
```

Comment:
```suggestion
    std::unique_ptr<RdKafka::Conf> Conf;
```
Do not use raw pointer.